src/custom-hooks/useAuthHandler.js

```diff
@@ -0,0 +1,176 @@
+import { useEffect, useReducer, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+import { useUser } from "context";
+import { useCookieHandler } from "custom-hooks";
+import {
+  ROUTE_ROOT,
+  API_TO_POST_SIGN_IN_DETAILS,
+  API_TO_POST_SIGN_UP_DETAILS,
+  checkAlphaNumericString,
+} from "utils";
+import {
+  authInitialReducerState,
+  authReducer,
+  ACTION_TYPE_ENTER_FORM_DETAILS,
+  ACTION_TYPE_ERROR,
+  ACTION_TYPE_LOADING,
+  ACTION_TYPE_SUCCESS,
+} from "reducer";
+
+/**
+ * useAuthHandler - hook
+ *
+ * @returns state and handler functions
+ *          related to authentication
+ */
```

Comment:
Appreciation - use of comments to signify what this hook does before its definition.